src/oauth2-client/controller/collection.ts

```diff
@@ -73,8 +73,18 @@ class ClientCollectionController extends Controller {
       requirePkce: ctx.request.body.requirePkce ?? false,
     };
 
-    const client = await create(newClient, redirectUris);
-    ctx.response.body = hal.newClientSuccess(client, redirectUris, clientSecret);
+    // If client id already exists in DB, throw error.
+    try {
+      const client = await create(newClient, redirectUris);
+      ctx.response.body = hal.newClientSuccess(client, redirectUris, clientSecret);
+    } catch (error: any) {
+      const err = wrapError(error);
```

Comment:
Awesome solution and great library.

A last thing I would change is to move this entire check to inside the service. The advantage is that if other code uses this `create` function, they just get nice errors.

Ideally the controller doesn't have to 'know' about databases. From a controllers perspective the service is a black box and the controller doesn't know if the service uses a database to store its information or not.